### ISSUE TITLE
elons-toys:  Inconsistent object instantiations in tests

### ISSUE THREAD

**Issue by Jarak-Jakar** [exercism/csharp#2167]
The first task in the Elon's Toys exercise is to create a `buy()` method that instantiates a new `RemoteControlCar` with full battery and 0 metres driven on it.  There's nothing stated about writing any other constructor, however.

The first three tests do indeed test that the `buy()` method behaves appropriately by creating a new car object with the method, but all subsequent tests just instantiate a new object by calling the default constructor directly, bypassing the `buy()` method.  This causes anyone who hasn't overridden the implicit default constructor to fail most subsequent tests.

An example of a solution to the problem which I believe is valid according to the instructions, but which fails, is my first iteration:

```c#
using System;

class RemoteControlCar
{
    private const int MetresPerDrive = 20;

    private int _metresDriven;
    private int _batteryPercentage;

    public static RemoteControlCar Buy() =>
        new()
        {
            _metresDriven = 0,
            _batteryPercentage = 100
        };

    public string DistanceDisplay() => $"Driven {_metresDriven} meters";

    public string BatteryDisplay() => _batteryPercentage > 0 ? $"Battery at {_batteryPercentage}%" : "Battery empty";

    public void Drive()
    {
        if (_batteryPercentage > 0)
        {
            _metresDriven += MetresPerDrive;
            _batteryPercentage--;
        }
    }
}
```

Basically, the issue here to me is that there's no instruction telling people to do anything to constructors, but rather it is implied that the `buy` method is the appropriate equivalent.  After following the instructions, however, it turns out that this is not sufficient to make the tests pass.  I suggest either changing the later tests to call `RemoteControlCar.buy()` instead of using the constructor, or changing the instructions to state explicitly that people should override the default constructor.
